docs/01-app/03-api-reference/01-directives/use-cache-remote.mdx

```diff
@@ -44,222 +47,164 @@ const nextConfig = {
+- Cache less data when possible (JSX output instead of raw API responses) to reduce in-memory allocation
 
-export default async function ProductPage({
-  params,
-}: {
-  params: Promise<{ id: string }>
-}) {
-  const { id } = await params
+### When remote caching makes sense
 
-  return (
-    <div>
-      <ProductDetails id={id} />
-      <Suspense fallback={<div>Loading price...</div>}>
-        <ProductPrice productId={id} />
-      </Suspense>
-    </div>
-  )
-}
+There are compelling scenarios where `'use cache: remote'` is the right choice despite the tradeoffs:
 
-function ProductDetails({ id }: { id: string }) {
-  return <div>Product: {id}</div>
-}
+- **Rate-limited APIs**: Your upstream service (CMS, analytics, third-party API) has rate limits or request quotas that you risk hitting
+- **Backend scaling constraints**: Your backend can't scale to match your frontend's request volume
+- **Expensive operations**: Database queries or computations that are costly to run repeatedly
+- **Flaky or unreliable services**: External services that occasionally fail or have availability issues
 
-async function ProductPrice({ productId }: { productId: string }) {
-  // Calling connection() makes this component dynamic, preventing
-  // it from being included in the static shell. This ensures the price
-  // is always fetched at request time.
-  await connection()
+In these cases, the cost and latency of remote caching is justified by avoiding worse outcomes (rate limit errors, backend overload, high compute bills, or degraded user experience).
 
-  // Now we can cache the price in a remote cache handler.
-  // Regular 'use cache' would NOT work here because we're in a dynamic context.
-  const price = await getProductPrice(productId)
+> **Good to know:** `'use cache: remote'` is not a substitute for addressing fundamental performance or architectural issues in your backend. If your API is slow due to missing indexes, inefficient queries, or poor data modeling, caching will only mask the problem temporarily. Focus on fixing the root cause first, then use caching strategically to reduce load on a well-designed system.
 
-  return <div>Price: ${price}</div>
-}
+Another compelling reason to use remote caching is for multi-process serverless environments where each process has its own memory. `use cache` may not provide server-side cache hits outside the static shell. Remote caching solves this by providing a shared cache across all processes.
 
-async function getProductPrice(productId: string) {
-  'use cache: remote'
-  cacheTag(`product-price-${productId}`)
-  cacheLife({ expire: 3600 }) // 1 hour
+### Rendering behavior and performance
 
-  // This database query is cached and shared across all users
-  return db.products.getPrice(productId)
-}
-```
+When using `'use cache: remote'`, the cache handler lookup happens during rendering and **blocks** until the network request completes. This has different implications depending on where it's used:
```

Comment:
I think this might make it clearer that you need to fetch the data before rendering?

```suggestion
When using `'use cache: remote'`, the cache handler lookup happens before rendering and **blocks** until the network request completes. This has different implications depending on where it's used:
```